From 92000b380a385b7aa223266a00eb5cead2b3562b Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Fri, 11 Aug 2023 11:51:17 +0600
Subject: [PATCH 2/4] ARM: dts: renesas: r8a779f0: sync with linux

Use file from linux commit ab6affd8d52588e08c8a94081d17b4e713942775
(which is v5.10.41/rcar-5.1.7.rc9)

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 arch/arm/dts/r8a779f0.dtsi                    | 2411 ++++++++++++++++-
 .../dt-bindings/ipmmu/rcar-ipmmu-domains.h    |   38 +
 include/dt-bindings/ipmmu/rcar-ipmmu.h        |   15 +
 3 files changed, 2385 insertions(+), 79 deletions(-)
 create mode 100644 include/dt-bindings/ipmmu/rcar-ipmmu-domains.h
 create mode 100644 include/dt-bindings/ipmmu/rcar-ipmmu.h

diff --git a/arch/arm/dts/r8a779f0.dtsi b/arch/arm/dts/r8a779f0.dtsi
index 741ab62e71..bdcadcf1d7 100644
--- a/arch/arm/dts/r8a779f0.dtsi
+++ b/arch/arm/dts/r8a779f0.dtsi
@@ -8,7 +8,7 @@
 #include <dt-bindings/clock/r8a779f0-cpg-mssr.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/power/r8a779f0-sysc.h>
-
+#include <dt-bindings/ipmmu/rcar-ipmmu.h>
 / {
 	compatible = "renesas,r8a779f0";
 	#address-cells = <2>;
@@ -23,24 +23,212 @@
 		i2c5 = &i2c5;
 	};
 
+	cluster1_opp: opp_table10 {
+		compatible = "operating-points-v2";
+		opp-shared;
+		opp-500000000 {
+			opp-hz = /bits/ 64 <500000000>;
+			opp-microvolt = <880000>;
+			clock-latency-ns = <500000>;
+		};
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <880000>;
+			clock-latency-ns = <500000>;
+		};
+		opp-1000000000 {
+			opp-hz = /bits/ 64 <1000000000>;
+			opp-microvolt = <880000>;
+			clock-latency-ns = <500000>;
+		};
+		opp-1200000000 {
+			opp-hz = /bits/ 64 <1200000000>;
+			opp-microvolt = <880000>;
+			clock-latency-ns = <500000>;
+			opp-suspend;
+		};
+	};
+
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		cpu-map {
+			cluster0 {
+				core0 {
+					cpu = <&a55_0>;
+				};
+				core1 {
+					cpu = <&a55_1>;
+				};
+			};
+			cluster1 {
+				core0 {
+					cpu = <&a55_2>;
+				};
+				core1 {
+					cpu = <&a55_3>;
+				};
+			};
+			cluster2 {
+				core0 {
+					cpu = <&a55_4>;
+				};
+				core1 {
+					cpu = <&a55_5>;
+				};
+			};
+			cluster3 {
+				core0 {
+					cpu = <&a55_6>;
+				};
+				core1 {
+					cpu = <&a55_7>;
+				};
+			};
+		};
+
 		a55_0: cpu@0 {
 			compatible = "arm,cortex-a55";
 			reg = <0>;
 			device_type = "cpu";
 			power-domains = <&sysc R8A779F0_PD_A1E0D0C0>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z0>;
+			operating-points-v2 = <&cluster1_opp>;
+			next-level-cache = <&L3_CA55_0>;
+		};
+
+		a55_1: cpu@100 {
+			compatible = "arm,cortex-a55";
+			reg = <0x100>;
+			device_type = "cpu";
+			power-domains = <&sysc R8A779F0_PD_A1E0D0C1>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z0>;
+			operating-points-v2 = <&cluster1_opp>;
 			next-level-cache = <&L3_CA55_0>;
 		};
 
+		a55_2: cpu@10000 {
+			compatible = "arm,cortex-a55";
+			reg = <0x10000>;
+			device_type = "cpu";
+			power-domains = <&sysc R8A779F0_PD_A1E0D1C0>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z0>;
+			operating-points-v2 = <&cluster1_opp>;
+			next-level-cache = <&L3_CA55_1>;
+		};
+
+		a55_3: cpu@10100 {
+			compatible = "arm,cortex-a55";
+			reg = <0x10100>;
+			device_type = "cpu";
+			power-domains = <&sysc R8A779F0_PD_A1E0D1C1>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z0>;
+			operating-points-v2 = <&cluster1_opp>;
+			next-level-cache = <&L3_CA55_1>;
+		};
+
+		a55_4: cpu@20000 {
+			compatible = "arm,cortex-a55";
+			reg = <0x20000>;
+			device_type = "cpu";
+			power-domains = <&sysc R8A779F0_PD_A1E1D0C0>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z1>;
+			operating-points-v2 = <&cluster1_opp>;
+			next-level-cache = <&L3_CA55_2>;
+		};
+
+		a55_5: cpu@20100 {
+			compatible = "arm,cortex-a55";
+			reg = <0x20100>;
+			device_type = "cpu";
+			power-domains = <&sysc R8A779F0_PD_A1E1D0C1>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z1>;
+			operating-points-v2 = <&cluster1_opp>;
+			next-level-cache = <&L3_CA55_2>;
+		};
+
+		a55_6: cpu@30000 {
+			compatible = "arm,cortex-a55";
+			reg = <0x30000>;
+			device_type = "cpu";
+			power-domains = <&sysc R8A779F0_PD_A1E1D1C0>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z1>;
+			operating-points-v2 = <&cluster1_opp>;
+			next-level-cache = <&L3_CA55_3>;
+		};
+
+		a55_7: cpu@30100 {
+			compatible = "arm,cortex-a55";
+			reg = <0x30100>;
+			device_type = "cpu";
+			power-domains = <&sysc R8A779F0_PD_A1E1D1C1>;
+			enable-method = "psci";
+			cpu-idle-states = <&CPU_SLEEP_0>;
+			clocks =<&cpg CPG_CORE R8A779F0_CLK_Z1>;
+			operating-points-v2 = <&cluster1_opp>;
+			next-level-cache = <&L3_CA55_3>;
+		};
+
+		idle-states {
+			entry-method = "psci";
+
+			CPU_SLEEP_0: cpu-sleep-0 {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x0010000>;
+				local-timer-stop;
+				entry-latency-us = <400>;
+				exit-latency-us = <500>;
+				min-residency-us = <4000>;
+			};
+	       };
+
 		L3_CA55_0: cache-controller-0 {
 			compatible = "cache";
 			power-domains = <&sysc R8A779F0_PD_A2E0D0>;
 			cache-unified;
 			cache-level = <3>;
 		};
+
+		L3_CA55_1: cache-controller-1 {
+			compatible = "cache";
+			power-domains = <&sysc R8A779F0_PD_A2E0D1>;
+			cache-unified;
+			cache-level = <3>;
+		};
+
+		L3_CA55_2: cache-controller-2 {
+			compatible = "cache";
+			power-domains = <&sysc R8A779F0_PD_A2E1D0>;
+			cache-unified;
+			cache-level = <3>;
+		};
+
+		L3_CA55_3: cache-controller-3 {
+			compatible = "cache";
+			power-domains = <&sysc R8A779F0_PD_A2E1D1>;
+			cache-unified;
+			cache-level = <3>;
+		};
+	};
+
+	psci {
+		compatible = "arm,psci-0.2";
+		method = "smc";
 	};
 
 	extal_clk: extal {
@@ -57,11 +245,32 @@
 		clock-frequency = <0>;
 	};
 
+	ufs30_clk: ufs30_refclk_v {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		/* This value must be overridden by the board */
+		clock-frequency = <0>;
+	};
+
+	firmware {
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+	};
+
 	pmu_a76 {
 		compatible = "arm,cortex-a55-pmu";
 		interrupts-extended = <&gic GIC_PPI 7 IRQ_TYPE_LEVEL_LOW>;
 	};
 
+	/* External PCIe clock - can be overridden by the board */
+	pcie_bus_clk: pcie_bus {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <0>;
+	};
+
 	/* External SCIF clock - to be overridden by boards that provide it */
 	scif_clk: scif {
 		compatible = "fixed-clock";
@@ -90,8 +299,8 @@
 			compatible = "renesas,pfc-r8a779f0";
 			reg = <0 0xe6050000 0 0x16c>, <0 0xe6050800 0 0x16c>,
 			      <0 0xe6051000 0 0x16c>, <0 0xe6051800 0 0x16c>,
-			      <0 0xffd90000 0 0x16c>, <0 0xffd90800 0 0x16c>,
-			      <0 0xffd91000 0 0x16c>, <0 0xffd91800 0 0x16c>;
+			      <0 0xdfd90000 0 0x16c>, <0 0xdfd90800 0 0x16c>,
+			      <0 0xdfd91000 0 0x16c>, <0 0xdfd91800 0 0x16c>;
 		};
 
 		gpio0: gpio@e6050180 {
@@ -150,62 +359,6 @@
 			resets = <&cpg 915>;
 		};
 
-		gpio4: gpio@ffd90180 {
-			compatible = "renesas,gpio-r8a779f0";
-			reg = <0 0xffd90180 0 0x54>;
-			interrupts = <GIC_SPI 826 IRQ_TYPE_LEVEL_HIGH>;
-			#gpio-cells = <2>;
-			gpio-controller;
-			gpio-ranges = <&pfc 0 128 31>;
-			#interrupt-cells = <2>;
-			interrupt-controller;
-			clocks = <&cpg CPG_MOD 915>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			resets = <&cpg 915>;
-		};
-
-		gpio5: gpio@ffd90980 {
-			compatible = "renesas,gpio-r8a779f0";
-			reg = <0 0xffd90980 0 0x54>;
-			interrupts = <GIC_SPI 827 IRQ_TYPE_LEVEL_HIGH>;
-			#gpio-cells = <2>;
-			gpio-controller;
-			gpio-ranges = <&pfc 0 160 20>;
-			#interrupt-cells = <2>;
-			interrupt-controller;
-			clocks = <&cpg CPG_MOD 915>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			resets = <&cpg 915>;
-		};
-
-		gpio6: gpio@ffd91180 {
-			compatible = "renesas,gpio-r8a779f0";
-			reg = <0 0xffd91180 0 0x54>;
-			interrupts = <GIC_SPI 828 IRQ_TYPE_LEVEL_HIGH>;
-			#gpio-cells = <2>;
-			gpio-controller;
-			gpio-ranges = <&pfc 0 192 32>;
-			#interrupt-cells = <2>;
-			interrupt-controller;
-			clocks = <&cpg CPG_MOD 915>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			resets = <&cpg 915>;
-		};
-
-		gpio7: gpio@ffd91980 {
-			compatible = "renesas,gpio-r8a779f0";
-			reg = <0 0xffd91980 0 0x54>;
-			interrupts = <GIC_SPI 829 IRQ_TYPE_LEVEL_HIGH>;
-			#gpio-cells = <2>;
-			gpio-controller;
-			gpio-ranges = <&pfc 0 224 32>;
-			#interrupt-cells = <2>;
-			interrupt-controller;
-			clocks = <&cpg CPG_MOD 915>;
-			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
-			resets = <&cpg 915>;
-		};
-
 		cpg: clock-controller@e6150000 {
 			compatible = "renesas,r8a779f0-cpg-mssr";
 			reg = <0 0xe6150000 0 0x4000>;
@@ -235,7 +388,9 @@
 			clocks = <&cpg CPG_MOD 518>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 518>;
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x91>, <&dmac0 0x90>,
+			       <&dmac1 0x91>, <&dmac1 0x90>;
+			dma-names = "tx", "rx", "tx", "rx";
 			i2c-scl-internal-delay-ns = <110>;
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -250,7 +405,9 @@
 			clocks = <&cpg CPG_MOD 519>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 519>;
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x93>, <&dmac0 0x92>,
+			       <&dmac1 0x93>, <&dmac1 0x92>;
+			dma-names = "tx", "rx", "tx", "rx";
 			i2c-scl-internal-delay-ns = <110>;
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -265,7 +422,9 @@
 			clocks = <&cpg CPG_MOD 520>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 520>;
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x95>, <&dmac0 0x94>,
+			       <&dmac1 0x95>, <&dmac1 0x94>;
+			dma-names = "tx", "rx", "tx", "rx";
 			i2c-scl-internal-delay-ns = <110>;
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -280,7 +439,9 @@
 			clocks = <&cpg CPG_MOD 521>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 521>;
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x97>, <&dmac0 0x96>,
+			       <&dmac1 0x97>, <&dmac1 0x96>;
+			dma-names = "tx", "rx", "tx", "rx";
 			i2c-scl-internal-delay-ns = <110>;
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -295,7 +456,9 @@
 			clocks = <&cpg CPG_MOD 522>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 522>;
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x99>, <&dmac0 0x98>,
+			       <&dmac1 0x99>, <&dmac1 0x98>;
+			dma-names = "tx", "rx", "tx", "rx";
 			i2c-scl-internal-delay-ns = <110>;
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -310,13 +473,28 @@
 			clocks = <&cpg CPG_MOD 523>;
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 523>;
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x9b>, <&dmac0 0x9a>,
+			       <&dmac1 0x9b>, <&dmac1 0x9a>;
+			dma-names = "tx", "rx", "tx", "rx";
 			i2c-scl-internal-delay-ns = <110>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 			status = "disabled";
 		};
 
+		tsc: thermal@e6198000 {
+			compatible = "renesas,r8a779f0-thermal";
+			reg = <0 0xe6198000 0 0x100>,
+			      <0 0xe61a0000 0 0x100>,
+			      <0 0xe61a8000 0 0x100>,
+			      <0 0xe61b0000 0 0x100>;
+			interrupts = <GIC_SPI 768 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 919>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 919>;
+			#thermal-sensor-cells = <1>;
+		};
+
 		hscif0: serial@e6540000 {
 			compatible = "renesas,hscif-r8a779f0",
 				     "renesas,rcar-gen4-hscif", "renesas,hscif";
@@ -326,7 +504,9 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x31>, <&dmac0 0x30>,
+			       <&dmac1 0x31>, <&dmac1 0x30>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 514>;
 			status = "disabled";
@@ -341,7 +521,9 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x33>, <&dmac0 0x32>,
+			       <&dmac1 0x33>, <&dmac1 0x32>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 515>;
 			status = "disabled";
@@ -356,7 +538,9 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x35>, <&dmac0 0x34>,
+			       <&dmac1 0x35>, <&dmac1 0x34>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 516>;
 			status = "disabled";
@@ -371,7 +555,9 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x37>, <&dmac0 0x36>,
+			       <&dmac1 0x37>, <&dmac1 0x36>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 517>;
 			status = "disabled";
@@ -383,11 +569,44 @@
 			reg = <0 0xe68c0000 0 0x00020000>,
 			      <0 0xe6444000 0 0x2800>;
 			reg-names = "iobase", "serdes";
+			interrupts = <GIC_SPI 280 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 282 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 283 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 284 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 285 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 286 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 287 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 288 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 289 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 290 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 291 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 292 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 293 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 294 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 295 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 296 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 297 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 298 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 299 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "gwca0_rxtx0", "gwca0_rxtx1",
+					  "gwca0_rxtx2", "gwca0_rxtx3",
+					  "gwca0_rxtx4", "gwca0_rxtx5",
+					  "gwca0_rxtx6", "gwca0_rxtx7",
+					  "gwca1_rxtx0", "gwca1_rxtx1",
+					  "gwca1_rxtx2", "gwca1_rxtx3",
+					  "gwca1_rxtx4", "gwca1_rxtx5",
+					  "gwca1_rxtx6", "gwca1_rxtx7",
+					  "gwca0_rxts0", "gwca0_rxts1",
+					  "gwca1_rxts0", "gwca1_rxts1";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			clocks = <&cpg CPG_MOD 1505>,
-				 <&cpg CPG_MOD 1506>;
+				<&cpg CPG_MOD 1506>;
 			clock-names = "rsw2", "eth-phy";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 1505>, <&cpg 1506>;
+			reset-names = "rsw2", "eth-phy";
 			status = "disabled";
 		};
 
@@ -400,7 +619,9 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x51>, <&dmac0 0x50>,
+			       <&dmac1 0x51>, <&dmac1 0x50>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 702>;
 			status = "disabled";
@@ -415,7 +636,9 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x53>, <&dmac0 0x52>,
+			       <&dmac1 0x53>, <&dmac1 0x52>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 703>;
 			status = "disabled";
@@ -430,7 +653,9 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x57>, <&dmac0 0x56>,
+			       <&dmac1 0x57>, <&dmac1 0x56>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 704>;
 			status = "disabled";
@@ -445,12 +670,158 @@
 				 <&cpg CPG_CORE R8A779F0_CLK_SASYNCPERD1>,
 				 <&scif_clk>;
 			clock-names = "fck", "brg_int", "scif_clk";
-			/* dmas and dma-names placeholder */
+			dmas = <&dmac0 0x59>, <&dmac0 0x58>,
+			       <&dmac1 0x59>, <&dmac1 0x58>;
+			dma-names = "tx", "rx", "tx", "rx";
 			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 			resets = <&cpg 705>;
 			status = "disabled";
 		};
 
+		msiof0: spi@e6e90000 {
+			compatible = "renesas,msiof-r8a779f0",
+				     "renesas,rcar-gen4-msiof";
+			reg = <0 0xe6e90000 0 0x0064>;
+			interrupts = <GIC_SPI 230 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 618>, <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			clock-names = "msiof_clk", "mso";
+			assigned-clocks = <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			assigned-clock-rates = <20000000>;
+			dmas = <&dmac0 0x41>, <&dmac0 0x40>,
+			       <&dmac1 0x41>, <&dmac1 0x40>;
+			dma-names = "tx", "rx", "tx", "rx";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 618>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+
+		msiof1: spi@e6ea0000 {
+			compatible = "renesas,msiof-r8a779f0",
+				     "renesas,rcar-gen4-msiof";
+			reg = <0 0xe6ea0000 0 0x0064>;
+			interrupts = <GIC_SPI 231 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 619>, <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			clock-names = "msiof_clk", "mso";
+			assigned-clocks = <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			assigned-clock-rates = <20000000>;
+			dmas = <&dmac0 0x43>, <&dmac0 0x42>,
+			       <&dmac1 0x43>, <&dmac1 0x42>;
+			dma-names = "tx", "rx", "tx", "rx";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 619>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+
+		msiof2: spi@e6c00000 {
+			compatible = "renesas,msiof-r8a779f0",
+				     "renesas,rcar-gen4-msiof";
+			reg = <0 0xe6c00000 0 0x0064>;
+			interrupts = <GIC_SPI 232 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 620>, <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			clock-names = "msiof_clk", "mso";
+			assigned-clocks = <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			assigned-clock-rates = <20000000>;
+			dmas = <&dmac0 0x45>, <&dmac0 0x44>,
+			       <&dmac1 0x45>, <&dmac1 0x44>;
+			dma-names = "tx", "rx", "tx", "rx";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 620>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+
+		msiof3: spi@e6c10000 {
+			compatible = "renesas,msiof-r8a779f0",
+				     "renesas,rcar-gen4-msiof";
+			reg = <0 0xe6c10000 0 0x0064>;
+			interrupts = <GIC_SPI 233 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 621>, <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			clock-names = "msiof_clk", "mso";
+			assigned-clocks = <&cpg CPG_CORE R8A779F0_CLK_MSO>;
+			assigned-clock-rates = <20000000>;
+			dmas = <&dmac0 0x47>, <&dmac0 0x46>,
+			       <&dmac1 0x47>, <&dmac1 0x46>;
+			dma-names = "tx", "rx", "tx", "rx";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 621>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "disabled";
+		};
+
+		dmac0: dma-controller@e7350000 {
+			compatible = "renesas,dmac-r8a779f0";
+			reg = <0 0xe7350000 0 0x1000>,
+			      <0 0xe7300000 0 0x10000>;
+			interrupts = <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 102 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 103 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 104 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 108 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 109 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 110 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 111 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 112 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3", "ch4",
+					  "ch5", "ch6", "ch7", "ch8", "ch9",
+					  "ch10", "ch11", "ch12", "ch13",
+					  "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 709>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 709>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
+		};
+
+		dmac1: dma-controller@e7351000 {
+			compatible = "renesas,dmac-r8a779f0";
+			reg = <0 0xe7351000 0 0x1000>,
+			      <0 0xe7310000 0 0x10000>;
+			interrupts = <GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 120 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 122 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 123 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 124 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 125 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 147 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 151 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 152 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 153 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 154 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3", "ch4",
+					  "ch5", "ch6", "ch7", "ch8", "ch9",
+					  "ch10", "ch11", "ch12", "ch13",
+					  "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 710>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 710>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
+		};
+
 		mmc0: mmc@ee140000 {
 			compatible = "renesas,sdhi-r8a779f0",
 				     "renesas,rcar-gen4-sdhi";
@@ -471,20 +842,1902 @@
 			reg = <0x0 0xf1000000 0 0x20000>,
 			      <0x0 0xf1060000 0 0x110000>;
 			interrupts = <GIC_PPI 9
-				      (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_HIGH)>;
+				      (GIC_CPU_MASK_SIMPLE(8) | IRQ_TYPE_LEVEL_HIGH)>;
 		};
 
 		prr: chipid@fff00044 {
 			compatible = "renesas,prr";
 			reg = <0 0xfff00044 0 4>;
+			status = "disabled";
+		};
+
+		rt_dmac0: dma-controller@ffd60000 {
+			compatible = "renesas,dmac-r8a779f0";
+			reg = <0 0xffd60000 0 0x1000>,
+			      <0 0xffc10000 0 0x10000>;
+			interrupts = <GIC_SPI 29 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 30 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 31 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 33 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 35 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 36 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 37 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 38 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 39 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 40 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 42 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 43 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 44 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 45 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3", "ch4",
+					  "ch5", "ch6", "ch7", "ch8", "ch9",
+					  "ch10", "ch11", "ch12", "ch13",
+					  "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 630>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 630>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
+			status = "disabled";
+		};
+
+		rt_dmac1: dma-controller@ffd61000 {
+			compatible = "renesas,dmac-r8a779f0";
+			reg = <0 0xffd61000 0 0x1000>,
+			      <0 0xffc20000 0 0x10000>;
+			interrupts = <GIC_SPI 47 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 50 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 51 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 52 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 53 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 54 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 55 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 56 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 57 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 59 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 60 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 61 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 63 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3", "ch4",
+					  "ch5", "ch6", "ch7", "ch8", "ch9",
+					  "ch10", "ch11", "ch12", "ch13",
+					  "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 631>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 631>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
+			status = "disabled";
+		};
+
+		rt_dmac2: dma-controller@ffd62000 {
+			compatible = "renesas,dmac-r8a779f0";
+			reg = <0 0xffd62000 0 0x1000>,
+			      <0 0xffd70000 0 0x10000>;
+			interrupts = <GIC_SPI 65 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 66 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 68 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 70 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 71 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 72 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 78 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 81 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3", "ch4",
+					  "ch5", "ch6", "ch7", "ch8", "ch9",
+					  "ch10", "ch11", "ch12", "ch13",
+					  "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 700>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 700>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
+			status = "disabled";
+		};
+
+		rt_dmac3: dma-controller@ffd63000 {
+			compatible = "renesas,dmac-r8a779f0";
+			reg = <0 0xffd63000 0 0x1000>,
+			      <0 0xffd80000 0 0x10000>;
+			interrupts = <GIC_SPI 83 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 84 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 85 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 87 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 89 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 90 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 92 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 93 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 94 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 95 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 97 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 99 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "error",
+					  "ch0", "ch1", "ch2", "ch3", "ch4",
+					  "ch5", "ch6", "ch7", "ch8", "ch9",
+					  "ch10", "ch11", "ch12", "ch13",
+					  "ch14", "ch15";
+			clocks = <&cpg CPG_MOD 701>;
+			clock-names = "fck";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 701>;
+			#dma-cells = <1>;
+			dma-channels = <16>;
+			status = "disabled";
+		};
+
+		ipmmu_mm: iommu@eefc0000 {
+			compatible = "renesas,ipmmu-r8a779f0";
+			reg = <0 0xeefc0000 0 0x20000>;
+			interrupts = <GIC_SPI 21 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 22 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+		ipmmu_rt0: iommu@ee480000 {
+			compatible = "renesas,ipmmu-r8a779f0";
+			reg = <0 0xee480000 0 0x20000>;
+			renesas,ipmmu-main = <&ipmmu_mm 10>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+		ipmmu_rt1: iommu@ee4c0000 {
+                        compatible = "renesas,ipmmu-r8a779f0";
+                        reg = <0 0xee4c0000 0 0x20000>;
+                        renesas,ipmmu-main = <&ipmmu_mm 19>;
+                        power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+                        #iommu-cells = <1>;
+                        status = "disabled";
+                };
+
+		ipmmu_ds0: iommu@eed00000 {
+			compatible = "renesas,ipmmu-r8a779f0";
+			reg = <0 0xeed00000 0 0x20000>;
+			renesas,ipmmu-main = <&ipmmu_mm 0>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+		ipmmu_hc: iommu@eed40000 {
+			compatible = "renesas,ipmmu-r8a779f0";
+			reg = <0 0xeed40000 0 0x20000>;
+			renesas,ipmmu-main = <&ipmmu_mm 2>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+                qos@e67e0000 {
+                        compatible = "renesas,qos";
+                        reg = <0 0xe67e0000 0 0x10090>;
+			status = "disabled";
+                };
+
+		pciec0: pcie@e65d0000 {
+			compatible = "renesas,r8a779f0-pcie";
+			reg = <0 0xe65d0000 0 0x3000>,
+			      <0 0xe65d3000 0 0x2000>,
+			      <0 0xe65d5000 0 0x1200>,
+			      <0 0xe65d6200 0 0x0e00>,
+			      <0 0xe65d7000 0 0x1000>,
+			      <0 0xfe000000 0 0x10000>;
+			reg-names = "dbi",
+				    "atu",
+				    "dma",
+				    "app",
+				    "phy",
+				    "config";
+			#address-cells = <3>;
+			#size-cells = <2>;
+			bus-range = <0x00 0xff>;
+			device_type = "pci";
+				 /* downstream IO */
+			ranges =  <0x81000000 0 0x00000000 0 0xfe000000 0 0x00010000
+				 /* non-prefetchable memory */
+				  0x82000000 0 0x30000000 0 0x30000000 0 0x10000000>;
+				     /* Map all possible DDR as inbound ranges */
+			dma-ranges = <0x42000000 0 0x40000000 0 0x40000000 0 0x80000000>;
+			interrupts = <GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 417 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 418 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 419 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 420 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 421 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 422 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "msi", "dma", "err", "fatal",
+					  "nonfatal", "lp", "vndmsg";
+			#interrupt-cells = <1>;
+			interrupt-map-mask = <0 0 0 7>;
+			interrupt-map = <0 0 0 1 &gic GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH
+					 0 0 0 2 &gic GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH
+					 0 0 0 3 &gic GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH
+					 0 0 0 4 &gic GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 624>,
+				 <&pcie_bus_clk>;
+			clock-names = "pcie0",
+				      "pcie_bus";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 624>;
+			num-lanes = <2>;
+			snps,enable-cdm-check = "true";
+			max-link-speed = <4>;
+			status = "disabled";
+		};
+
+		pciec0_ep: pciec0_ep@e65d0000 {
+			compatible = "renesas,r8a779f0-pcie-ep";
+			reg = <0 0xe65d0000 0 0x1000>, <0 0xe65d1000 0 0x1000>,
+			      <0 0xe65d2000 0 0x1000>, <0 0xe65d3000 0 0x2000>,
+			      <0 0xe65d5000 0 0x1200>, <0 0xe65d6200 0 0x0e00>,
+			      <0 0xe65d7000 0 0x1000>, <0 0xfe000000 0 0x400000>;
+			reg-names = "dbi", "dbi2",
+				    "shadow", "atu",
+				    "dma", "app",
+				    "phy", "addr_space";
+			interrupts = <GIC_SPI 416 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 417 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 418 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 419 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 420 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 421 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 422 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "msi", "dma", "err", "fatal",
+					  "nonfatal", "lp", "vndmsg";
+			clocks = <&cpg CPG_MOD 624>,
+				 <&pcie_bus_clk>;
+			clock-names = "pcie0",
+				      "pcie_bus";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 624>;
+			num-lanes = <2>;
+			max-link-speed = <4>;
+			num-ib-windows = <16>;
+			num-ob-windows = <16>;
+			status = "disabled";
+		};
+
+		pciec1: pcie@e65d8000 {
+			compatible = "renesas,r8a779f0-pcie";
+			reg = <0 0xe65d8000 0 0x3000>,
+			      <0 0xe65db000 0 0x2000>,
+			      <0 0xe65dd000 0 0x1200>,
+			      <0 0xe65de200 0 0x0e00>,
+			      <0 0xe65df000 0 0x1000>,
+			      <0 0xee900000 0 0x10000>,
+			      <0 0xe65d6200 0 0x0e00>;
+			reg-names = "dbi",
+				    "atu",
+				    "dma",
+				    "app",
+				    "phy",
+				    "config",
+				    "shared";
+			#address-cells = <3>;
+			#size-cells = <2>;
+			bus-range = <0x00 0xff>;
+			device_type = "pci";
+				 /* downstream IO */
+			ranges = <0x81000000 0 0x00000000 0 0xee900000 0 0x00010000
+				 /* non-prefetchable memory */
+				 0x82000000 0 0xc0000000 0 0xc0000000 0 0x10000000>;
+				     /* Map all possible DDR as inbound ranges */
+			dma-ranges = <0x42000000 0 0x40000000 0 0x40000000 0 0x80000000>;
+			interrupts = <GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 424 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 425 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 426 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 427 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 428 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 429 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "msi", "dma", "err", "fatal",
+					  "nonfatal", "lp", "vndmsg";
+			#interrupt-cells = <1>;
+			interrupt-map-mask = <0 0 0 7>;
+			interrupt-map = <0 0 0 1 &gic GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH
+					 0 0 0 2 &gic GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH
+					 0 0 0 3 &gic GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH
+					 0 0 0 4 &gic GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 625>,
+				 <&pcie_bus_clk>,
+				 <&cpg CPG_MOD 624>;
+			clock-names = "pcie1",
+				      "pcie_bus",
+				      "shared";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 625>;
+			num-lanes = <2>;
+			snps,enable-cdm-check = "true";
+			max-link-speed = <4>;
+			status = "disabled";
+		};
+
+		pciec1_ep: pciec1_ep@e65d8000 {
+			compatible = "renesas,r8a779f0-pcie-ep";
+			reg = <0 0xe65d8000 0 0x1000>, <0 0xe65d9000 0 0x1000>,
+			      <0 0xe65da000 0 0x1000>, <0 0xe65db000 0 0x2000>,
+			      <0 0xe65dd000 0 0x1200>, <0 0xe65de200 0 0x0e00>,
+			      <0 0xe65df000 0 0x1000>, <0 0xee900000 0 0x400000>,
+			      <0 0xe65d6200 0 0x0e00>;
+			reg-names = "dbi", "dbi2",
+				    "shadow", "atu",
+				    "dma", "app",
+				    "phy", "addr_space",
+				    "shared";
+			interrupts = <GIC_SPI 423 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 424 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 425 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 426 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 427 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 428 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 429 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "msi", "dma", "err", "fatal",
+					  "nonfatal", "lp", "vndmsg";
+			clocks = <&cpg CPG_MOD 625>,
+				 <&pcie_bus_clk>,
+				 <&cpg CPG_MOD 624>;
+			clock-names = "pcie1",
+				      "pcie_bus",
+				      "shared";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 625>;
+			num-lanes = <2>;
+			max-link-speed = <4>;
+			num-ib-windows = <16>;
+			num-ob-windows = <16>;
+			status = "disabled";
+		};
+
+		tau_pwm0: tau@dfbf4000 {
+			compatible = "renesas,tau-pwm";
+			reg = <0 0xdfbf4000 0 0x1000>,
+			      <0 0xdf980000 0 0x10000>,
+			      <0 0xe6160000 0 0x8>;
+			reg-names = "taud", "clkc", "modemr";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			renesas,channel = <0>;
+			status = "disabled";
+		};
+
+		tau_pwm1: tau@dfbf5000 {
+			compatible = "renesas,tau-pwm";
+			reg = <0 0xdfbf5000 0 0x1000>,
+			      <0 0xdf980000 0 0x10000>,
+			      <0 0xe6160000 0 0x8>;
+			reg-names = "taud", "clkc", "modemr";
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			renesas,channel = <1>;
+			status = "disabled";
+		};
+
+		iccom_reg00: mfis@e6260000 {
+			compatible = "generic-uio";
+			reg = <0 0xe6260000 0 0x10000>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_reg_00";
+		};
+
+		iccom00 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 320 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_00";
+		};
+
+		iccom01 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 322 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_01";
+		};
+
+		iccom02 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 324 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_02";
+		};
+
+		iccom03 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 326 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_03";
+		};
+
+		iccom04 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 328 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_04";
+		};
+
+		iccom05 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 330 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_05";
+		};
+
+		iccom06 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 332 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_06";
+		};
+
+		iccom07 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 334 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_07";
+		};
+
+		iccom08 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 336 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_08";
+		};
+
+		iccom09 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 338 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_09";
+		};
+
+		iccom010 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 340 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_10";
+		};
+
+
+		iccom011 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 342 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_11";
+		};
+
+		iccom012 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 344 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_12";
+		};
+
+		iccom013 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 346 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_13";
+		};
+
+		iccom014 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 348 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_14";
+		};
+
+		iccom015 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 350 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_15";
+		};
+
+		iccom016 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 354 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_16";
+		};
+
+		iccom017 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 356 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_17";
+		};
+
+		iccom018 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 358 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_18";
+		};
+
+		iccom019 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 360 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_19";
+		};
+
+		iccom020 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 362 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_20";
+		};
+
+		iccom021 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 364 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_21";
+		};
+
+		iccom022 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 366 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_22";
+		};
+
+		iccom023 {
+			compatible = "generic-uio";
+			interrupts = <GIC_SPI 368 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "iccom_23";
+		};
+
+		caiplite_wrapper {
+			compatible = "generic-uio";
+			clocks = <&cpg CPG_MOD 720>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 720>;
+			linux,uio-name = "caiplite_wrapper_00";
+		};
+
+		caiplite0 {
+			compatible = "generic-uio";
+			reg = <0 0xe7800000 0 0x1000>;
+			interrupts = <GIC_SPI 933 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 721>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 721>;
+			linux,uio-name = "caiplite_00";
+		};
+
+		caiplite1 {
+			compatible = "generic-uio";
+			reg = <0 0xe7801000 0 0x1000>;
+			interrupts = <GIC_SPI 935 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 722>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 722>;
+			linux,uio-name = "caiplite_01";
+		};
+
+		caiplite2 {
+			compatible = "generic-uio";
+			reg = <0 0xe7802000 0 0x1000>;
+			interrupts = <GIC_SPI 937 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 723>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 723>;
+			linux,uio-name = "caiplite_02";
+		};
+
+		caiplite3 {
+			compatible = "generic-uio";
+			reg = <0 0xe7803000 0 0x1000>;
+			interrupts = <GIC_SPI 939 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 724>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 724>;
+			linux,uio-name = "caiplite_03";
+		};
+
+		caiplite4 {
+			compatible = "generic-uio";
+			reg = <0 0xe7804000 0 0x1000>;
+			interrupts = <GIC_SPI 941 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 725>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 725>;
+			linux,uio-name = "caiplite_04";
+		};
+
+		caiplite5 {
+			compatible = "generic-uio";
+			reg = <0 0xe7805000 0 0x1000>;
+			interrupts = <GIC_SPI 943 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 726>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 726>;
+			linux,uio-name = "caiplite_05";
+		};
+
+		caiplite6 {
+			compatible = "generic-uio";
+			reg = <0 0xe7806000 0 0x1000>;
+			interrupts = <GIC_SPI 945 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 727>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 727>;
+			linux,uio-name = "caiplite_06";
+		};
+
+		caiplite7 {
+			compatible = "generic-uio";
+			reg = <0 0xe7807000 0 0x1000>;
+			interrupts = <GIC_SPI 947 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 728>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 728>;
+			linux,uio-name = "caiplite_07";
+		};
+
+		wcrc_caiplite0 {
+			compatible = "generic-uio";
+			reg = <0 0xe7080000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 903>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 903>;
+			linux,uio-name = "wcrc_caiplite_00";
+		};
+
+		wcrc_caiplite1 {
+			compatible = "generic-uio";
+			reg = <0 0xe7090000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 904>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 904>;
+			linux,uio-name = "wcrc_caiplite_01";
+		};
+
+		wcrc_caiplite2 {
+			compatible = "generic-uio";
+			reg = <0 0xe70a0000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 905>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 905>;
+			linux,uio-name = "wcrc_caiplite_02";
+		};
+
+		wcrc_caiplite3 {
+			compatible = "generic-uio";
+			reg = <0 0xe70b0000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 906>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 906>;
+			linux,uio-name = "wcrc_caiplite_03";
+		};
+
+		rtdmac_control0_00: rtdmac_control@ffc10000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc10000 0 0x1000>;
+			interrupts = <GIC_SPI 30 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_00";
+			ipmmu-id = <RCAR_IPMMU_RT1 0>;
+			iommus = <&ipmmu_rt1 0>;
+		};
+
+		rtdmac_control0_01: rtdmac_control@ffc11000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc11000 0 0x1000>;
+			interrupts = <GIC_SPI 31 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_01";
+			ipmmu-id = <RCAR_IPMMU_RT1 1>;
+			iommus = <&ipmmu_rt1 1>;
+		};
+
+		rtdmac_control0_02: rtdmac_control@ffc12000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc12000 0 0x1000>;
+			interrupts = <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_02";
+			ipmmu-id = <RCAR_IPMMU_RT1 2>;
+			iommus = <&ipmmu_rt1 2>;
+		};
+
+		rtdmac_control0_03: rtdmac_control@ffc13000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc13000 0 0x1000>;
+			interrupts = <GIC_SPI 33 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_03";
+			ipmmu-id = <RCAR_IPMMU_RT1 3>;
+			iommus = <&ipmmu_rt1 3>;
+		};
+
+		rtdmac_control0_04: rtdmac_control@ffc14000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc14000 0 0x1000>;
+			interrupts = <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_04";
+			ipmmu-id = <RCAR_IPMMU_RT1 4>;
+			iommus = <&ipmmu_rt1 4>;
+		};
+
+		rtdmac_control0_05: rtdmac_control@ffc15000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc15000 0 0x1000>;
+			interrupts = <GIC_SPI 35 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			ipmmu-id = <RCAR_IPMMU_RT1 5>;
+			linux,uio-name = "rtdmac_control0_05";
+			iommus = <&ipmmu_rt1 5>;
+		};
+
+		rtdmac_control0_06: rtdmac_control@ffc16000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc16000 0 0x1000>;
+			interrupts = <GIC_SPI 36 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_06";
+			ipmmu-id = <RCAR_IPMMU_RT1 6>;
+			iommus = <&ipmmu_rt1 6>;
+		};
+
+		rtdmac_control0_07: rtdmac_control@ffc17000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc17000 0 0x1000>;
+			interrupts = <GIC_SPI 37 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_07";
+			ipmmu-id = <RCAR_IPMMU_RT1 7>;
+			iommus = <&ipmmu_rt1 7>;
+		};
+
+		rtdmac_control0_08: rtdmac_control@ffc18000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc18000 0 0x1000>;
+			interrupts = <GIC_SPI 38 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_08";
+			ipmmu-id = <RCAR_IPMMU_RT1 8>;
+			iommus = <&ipmmu_rt1 8>;
+		};
+
+		rtdmac_control0_09: rtdmac_control@ffc19000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc19000 0 0x1000>;
+			interrupts = <GIC_SPI 39 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_09";
+			ipmmu-id = <RCAR_IPMMU_RT1 9>;
+			iommus = <&ipmmu_rt1 9>;
+		};
+
+		rtdmac_control0_10: rtdmac_control@ffc1a000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc1a000 0 0x1000>;
+			interrupts = <GIC_SPI 40 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_10";
+			ipmmu-id = <RCAR_IPMMU_RT1 10>;
+			iommus = <&ipmmu_rt1 10>;
+		};
+
+		rtdmac_control0_11: rtdmac_control@ffc1b000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc1b000 0 0x1000>;
+			interrupts = <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_11";
+			ipmmu-id = <RCAR_IPMMU_RT1 11>;
+			iommus = <&ipmmu_rt1 11>;
+		};
+
+		rtdmac_control0_12: rtdmac_control@ffc1c000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc1c000 0 0x1000>;
+			interrupts = <GIC_SPI 42 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_12";
+			ipmmu-id = <RCAR_IPMMU_RT1 12>;
+			iommus = <&ipmmu_rt1 12>;
+		};
+
+		rtdmac_control0_13: rtdmac_control@ffc1d000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc1d000 0 0x1000>;
+			interrupts = <GIC_SPI 43 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			ipmmu-id = <RCAR_IPMMU_RT1 13>;
+			linux,uio-name = "rtdmac_control0_13";
+			iommus = <&ipmmu_rt1 13>;
+		};
+
+		rtdmac_control0_14: rtdmac_control@ffc1e000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc1e000 0 0x1000>;
+			interrupts = <GIC_SPI 44 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			ipmmu-id = <RCAR_IPMMU_RT1 14>;
+			linux,uio-name = "rtdmac_control0_14";
+			iommus = <&ipmmu_rt1 14>;
+		};
+
+		rtdmac_control0_15: rtdmac_control@ffc1f000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc1f000 0 0x1000>;
+			interrupts = <GIC_SPI 45 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control0_15";
+			ipmmu-id = <RCAR_IPMMU_RT1 15>;
+			iommus = <&ipmmu_rt1 15>;
+		};
+
+		rtdmac_control1_00: rtdmac_control@ffc20000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc20000 0 0x1000>;
+			interrupts = <GIC_SPI 48 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_00";
+			ipmmu-id = <RCAR_IPMMU_RT1 16>;
+			iommus = <&ipmmu_rt1 16>;
+		};
+
+		rtdmac_control1_01: rtdmac_control@ffc21000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc21000 0 0x1000>;
+			interrupts = <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_01";
+			ipmmu-id = <RCAR_IPMMU_RT1 17>;
+			iommus = <&ipmmu_rt1 17>;
+		};
+
+		rtdmac_control1_02: rtdmac_control@ffc22000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc22000 0 0x1000>;
+			interrupts = <GIC_SPI 50 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_02";
+			ipmmu-id = <RCAR_IPMMU_RT1 18>;
+			iommus = <&ipmmu_rt1 18>;
+		};
+
+		rtdmac_control1_03: rtdmac_control@ffc23000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc23000 0 0x1000>;
+			interrupts = <GIC_SPI 51 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_03";
+			ipmmu-id = <RCAR_IPMMU_RT1 19>;
+			iommus = <&ipmmu_rt1 19>;
+		};
+
+		rtdmac_control1_04: rtdmac_control@ffc24000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc24000 0 0x1000>;
+			interrupts = <GIC_SPI 52 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_04";
+			ipmmu-id = <RCAR_IPMMU_RT1 20>;
+			iommus = <&ipmmu_rt1 20>;
+		};
+
+		rtdmac_control1_05: rtdmac_control@ffc25000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc25000 0 0x1000>;
+			interrupts = <GIC_SPI 53 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_05";
+			ipmmu-id = <RCAR_IPMMU_RT1 21>;
+			iommus = <&ipmmu_rt1 21>;
+		};
+
+		rtdmac_control1_06: rtdmac_control@ffc26000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc26000 0 0x1000>;
+			interrupts = <GIC_SPI 54 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_06";
+			ipmmu-id = <RCAR_IPMMU_RT1 22>;
+			iommus = <&ipmmu_rt1 22>;
+		};
+
+		rtdmac_control1_07: rtdmac_control@ffc27000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc27000 0 0x1000>;
+			interrupts = <GIC_SPI 55 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_07";
+			ipmmu-id = <RCAR_IPMMU_RT1 23>;
+			iommus = <&ipmmu_rt1 23>;
+		};
+
+		rtdmac_control1_08: rtdmac_control@ffc28000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc28000 0 0x1000>;
+			interrupts = <GIC_SPI 56 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_08";
+			ipmmu-id = <RCAR_IPMMU_RT1 24>;
+			iommus = <&ipmmu_rt1 24>;
+		};
+
+		rtdmac_control1_09: rtdmac_control@ffc29000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc29000 0 0x1000>;
+			interrupts = <GIC_SPI 57 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_09";
+			ipmmu-id = <RCAR_IPMMU_RT1 25>;
+			iommus = <&ipmmu_rt1 25>;
+		};
+
+		rtdmac_control1_10: rtdmac_control@ffc2a000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc2a000 0 0x1000>;
+			interrupts = <GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_10";
+			ipmmu-id = <RCAR_IPMMU_RT1 26>;
+			iommus = <&ipmmu_rt1 26>;
+		};
+
+		rtdmac_control1_11: rtdmac_control@ffc2b000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc2b000 0 0x1000>;
+			interrupts = <GIC_SPI 59 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_11";
+			ipmmu-id = <RCAR_IPMMU_RT1 27>;
+			iommus = <&ipmmu_rt1 27>;
+		};
+
+		rtdmac_control1_12: rtdmac_control@ffc2c000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc2c000 0 0x1000>;
+			interrupts = <GIC_SPI 60 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_12";
+			ipmmu-id = <RCAR_IPMMU_RT1 28>;
+			iommus = <&ipmmu_rt1 28>;
+		};
+
+		rtdmac_control1_13: rtdmac_control@ffc2d000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc2d000 0 0x1000>;
+			interrupts = <GIC_SPI 61 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_13";
+			ipmmu-id = <RCAR_IPMMU_RT1 29>;
+			iommus = <&ipmmu_rt1 29>;
+		};
+
+		rtdmac_control1_14: rtdmac_control@ffc2e000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc2e000 0 0x1000>;
+			interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_14";
+			ipmmu-id = <RCAR_IPMMU_RT1 30>;
+			iommus = <&ipmmu_rt1 30>;
+		};
+
+		rtdmac_control1_15: rtdmac_control@ffc2f000 {
+			compatible = "generic-uio";
+			reg = <0 0xffc2f000 0 0x1000>;
+			interrupts = <GIC_SPI 63 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control1_15";
+			ipmmu-id = <RCAR_IPMMU_RT1 31>;
+			iommus = <&ipmmu_rt1 31>;
+		};
+
+		rtdmac_control2_00: rtdmac_control@ffd70000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd70000 0 0x1000>;
+			interrupts = <GIC_SPI 66 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_00";
+			ipmmu-id = <RCAR_IPMMU_RT1 32>;
+			iommus = <&ipmmu_rt1 32>;
+		};
+
+		rtdmac_control2_01: rtdmac_control@ffd71000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd71000 0 0x1000>;
+			interrupts = <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_01";
+			ipmmu-id = <RCAR_IPMMU_RT1 33>;
+			iommus = <&ipmmu_rt1 33>;
+		};
+
+		rtdmac_control2_02: rtdmac_control@ffd72000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd72000 0 0x1000>;
+			interrupts = <GIC_SPI 68 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_02";
+			ipmmu-id = <RCAR_IPMMU_RT1 34>;
+			iommus = <&ipmmu_rt1 34>;
+		};
+
+		rtdmac_control2_03: rtdmac_control@ffd73000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd73000 0 0x1000>;
+			interrupts = <GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_03";
+			ipmmu-id = <RCAR_IPMMU_RT1 35>;
+			iommus = <&ipmmu_rt1 35>;
+		};
+
+		rtdmac_control2_04: rtdmac_control@ffd74000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd74000 0 0x1000>;
+			interrupts = <GIC_SPI 70 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_04";
+			ipmmu-id = <RCAR_IPMMU_RT1 36>;
+			iommus = <&ipmmu_rt1 36>;
+		};
+
+		rtdmac_control2_05: rtdmac_control@ffd75000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd75000 0 0x1000>;
+			interrupts = <GIC_SPI 71 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_05";
+			ipmmu-id = <RCAR_IPMMU_RT1 37>;
+			iommus = <&ipmmu_rt1 37>;
+		};
+
+		rtdmac_control2_06: rtdmac_control@ffd76000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd76000 0 0x1000>;
+			interrupts = <GIC_SPI 72 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_06";
+			ipmmu-id = <RCAR_IPMMU_RT1 38>;
+			iommus = <&ipmmu_rt1 38>;
+		};
+
+		rtdmac_control2_07: rtdmac_control@ffd77000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd77000 0 0x1000>;
+			interrupts = <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_07";
+			ipmmu-id = <RCAR_IPMMU_RT1 39>;
+			iommus = <&ipmmu_rt1 39>;
+		};
+
+		rtdmac_control2_08: rtdmac_control@ffd78000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd78000 0 0x1000>;
+			interrupts = <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_08";
+			ipmmu-id = <RCAR_IPMMU_RT1 40>;
+			iommus = <&ipmmu_rt1 40>;
+		};
+
+		rtdmac_control2_09: rtdmac_control@ffd79000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd79000 0 0x1000>;
+			interrupts = <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_09";
+			ipmmu-id = <RCAR_IPMMU_RT1 41>;
+			iommus = <&ipmmu_rt1 41>;
+		};
+
+		rtdmac_control2_10: rtdmac_control@ffd7a000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd7a000 0 0x1000>;
+			interrupts = <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_10";
+			ipmmu-id = <RCAR_IPMMU_RT1 42>;
+			iommus = <&ipmmu_rt1 42>;
+		};
+
+		rtdmac_control2_11: rtdmac_control@ffd7b000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd7b000 0 0x1000>;
+			interrupts = <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_11";
+			ipmmu-id = <RCAR_IPMMU_RT1 43>;
+			iommus = <&ipmmu_rt1 43>;
+		};
+
+		rtdmac_control2_12: rtdmac_control@ffd7c000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd7c000 0 0x1000>;
+			interrupts = <GIC_SPI 78 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_12";
+			ipmmu-id = <RCAR_IPMMU_RT1 44>;
+			iommus = <&ipmmu_rt1 44>;
+		};
+
+		rtdmac_control2_13: rtdmac_control@ffd7d000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd7d000 0 0x1000>;
+			interrupts = <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_13";
+			ipmmu-id = <RCAR_IPMMU_RT1 45>;
+			iommus = <&ipmmu_rt1 45>;
+		};
+
+		rtdmac_control2_14: rtdmac_control@ffd7e000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd7e000 0 0x1000>;
+			interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_14";
+			ipmmu-id = <RCAR_IPMMU_RT1 46>;
+			iommus = <&ipmmu_rt1 46>;
+		};
+
+		rtdmac_control2_15: rtdmac_control@ffd7f000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd7f000 0 0x1000>;
+			interrupts = <GIC_SPI 81 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control2_15";
+			ipmmu-id = <RCAR_IPMMU_RT1 47>;
+			iommus = <&ipmmu_rt1 47>;
+		};
+
+		rtdmac_control3_00: rtdmac_control@ffd80000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd80000 0 0x1000>;
+			interrupts = <GIC_SPI 84 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_00";
+			ipmmu-id = <RCAR_IPMMU_RT1 48>;
+			iommus = <&ipmmu_rt1 48>;
+		};
+
+		rtdmac_control3_01: rtdmac_control@ffd81000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd81000 0 0x1000>;
+			interrupts = <GIC_SPI 85 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_01";
+			ipmmu-id = <RCAR_IPMMU_RT1 49>;
+			iommus = <&ipmmu_rt1 49>;
+		};
+
+		rtdmac_control3_02: rtdmac_control@ffd82000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd82000 0 0x1000>;
+			interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_02";
+			ipmmu-id = <RCAR_IPMMU_RT1 50>;
+			iommus = <&ipmmu_rt1 50>;
+		};
+
+		rtdmac_control3_03: rtdmac_control@ffd83000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd83000 0 0x1000>;
+			interrupts = <GIC_SPI 87 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_03";
+			ipmmu-id = <RCAR_IPMMU_RT1 51>;
+			iommus = <&ipmmu_rt1 51>;
+		};
+
+		rtdmac_control3_04: rtdmac_control@ffd84000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd84000 0 0x1000>;
+			interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_04";
+			ipmmu-id = <RCAR_IPMMU_RT1 52>;
+			iommus = <&ipmmu_rt1 52>;
+		};
+
+		rtdmac_control3_05: rtdmac_control@ffd85000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd85000 0 0x1000>;
+			interrupts = <GIC_SPI 89 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_05";
+			ipmmu-id = <RCAR_IPMMU_RT1 53>;
+			iommus = <&ipmmu_rt1 53>;
+		};
+
+		rtdmac_control3_06: rtdmac_control@ffd86000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd86000 0 0x1000>;
+			interrupts = <GIC_SPI 90 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_06";
+			ipmmu-id = <RCAR_IPMMU_RT1 54>;
+			iommus = <&ipmmu_rt1 54>;
+		};
+
+		rtdmac_control3_07: rtdmac_control@ffd87000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd87000 0 0x1000>;
+			interrupts = <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_07";
+			ipmmu-id = <RCAR_IPMMU_RT1 55>;
+			iommus = <&ipmmu_rt1 55>;
+		};
+
+		rtdmac_control3_08: rtdmac_control@ffd88000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd88000 0 0x1000>;
+			interrupts = <GIC_SPI 92 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_08";
+			ipmmu-id = <RCAR_IPMMU_RT1 56>;
+			iommus = <&ipmmu_rt1 56>;
+		};
+
+		rtdmac_control3_09: rtdmac_control@ffd89000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd89000 0 0x1000>;
+			interrupts = <GIC_SPI 93 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_09";
+			ipmmu-id = <RCAR_IPMMU_RT1 57>;
+			iommus = <&ipmmu_rt1 57>;
+		};
+
+		rtdmac_control3_10: rtdmac_control@ffd8a000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd8a000 0 0x1000>;
+			interrupts = <GIC_SPI 94 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_10";
+			ipmmu-id = <RCAR_IPMMU_RT1 58>;
+			iommus = <&ipmmu_rt1 58>;
+		};
+
+		rtdmac_control3_11: rtdmac_control@ffd8b000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd8b000 0 0x1000>;
+			interrupts = <GIC_SPI 95 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_11";
+			ipmmu-id = <RCAR_IPMMU_RT1 59>;
+			iommus = <&ipmmu_rt1 59>;
+		};
+
+		rtdmac_control3_12: rtdmac_control@ffd8c000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd8c000 0 0x1000>;
+			interrupts = <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_12";
+			ipmmu-id = <RCAR_IPMMU_RT1 60>;
+			iommus = <&ipmmu_rt1 60>;
+		};
+
+		rtdmac_control3_13: rtdmac_control@ffd8d000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd8d000 0 0x1000>;
+			interrupts = <GIC_SPI 97 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_13";
+			ipmmu-id = <RCAR_IPMMU_RT1 61>;
+			iommus = <&ipmmu_rt1 61>;
+		};
+
+		rtdmac_control3_14: rtdmac_control@ffd8e000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd8e000 0 0x1000>;
+			interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_14";
+			ipmmu-id = <RCAR_IPMMU_RT1 62>;
+			iommus = <&ipmmu_rt1 62>;
+		};
+
+		rtdmac_control3_15: rtdmac_control@ffd8f000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd8f000 0 0x1000>;
+			interrupts = <GIC_SPI 99 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rtdmac_control3_15";
+			ipmmu-id = <RCAR_IPMMU_RT1 63>;
+			iommus = <&ipmmu_rt1 63>;
+		};
+
+		rtdmac_descriptor0: rtdmac_descriptor@ffd60000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd60000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 630>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 630>;
+			linux,uio-name = "rtdmac_descriptor0_00";
+		};
+
+		rtdmac_descriptor1: rtdmac_descriptor@ffd61000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd61000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 631>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 631>;
+			linux,uio-name = "rtdmac_descriptor1_00";
+		};
+
+		rtdmac_descriptor2: rtdmac_descriptor@ffd62000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd62000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 700>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 700>;
+			linux,uio-name = "rtdmac_descriptor2_00";
+		};
+
+		rtdmac_descriptor3: rtdmac_descriptor@ffd63000 {
+			compatible = "generic-uio";
+			reg = <0 0xffd63000 0 0x1000>;
+			clocks = <&cpg CPG_MOD 701>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 701>;
+			linux,uio-name = "rtdmac_descriptor3_00";
+		};
+
+		ships0: ships0@e7810000 {
+			compatible = "generic-uio";
+			reg = <0 0xe7810000 0 0x10000>;
+			interrupts = <GIC_SPI 948 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 719>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 719>;
+			linux,uio-name = "ships_00";
+		};
+
+		ships0_01 {
+			compatible = "generic-uio";
+			reg = <0 0xe7810000 0 0x10000>;
+			interrupts = <GIC_SPI 949 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_ships_00_01";
+		};
+
+		ships1: ships1@e7820000 {
+			compatible = "generic-uio";
+			reg = <0 0xe7820000 0 0x10000>;
+			interrupts = <GIC_SPI 950 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 1511>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 1511>;
+			linux,uio-name = "ships_01";
+		};
+
+		ships1_01 {
+			compatible = "generic-uio";
+			reg = <0 0xe7820000 0 0x10000>;
+			interrupts = <GIC_SPI 951 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_ships_01_01";
+		};
+
+		ships2: ships2@e7830000 {
+			compatible = "generic-uio";
+			reg = <0 0xe7830000 0 0x10000>;
+			interrupts = <GIC_SPI 952 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 1512>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 1512>;
+			linux,uio-name = "ships_02";
+		};
+
+		ships2_01 {
+			compatible = "generic-uio";
+			reg = <0 0xe7830000 0 0x10000>;
+			interrupts = <GIC_SPI 953 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_ships_02_01";
+		};
+
+		systemram0: systemram@e6300000 {
+			compatible = "generic-uio";
+			reg = <0 0xe6300000 0 0x60000>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "system_ram_00";
+		};
+
+		icumx_cmd0: icumx_cmd@e6610000 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "icumx_cmd_00";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_01 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 193 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_01";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_02 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 194 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_02";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_03 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_03";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_04 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_04";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_05 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_05";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_06 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 198 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_06";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_07 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 199 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_07";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_08 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 200 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_08";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_09 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 201 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_09";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_10 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 202 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_10";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_11 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_11";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_12 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 204 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_12";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_13 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 205 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_13";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_14 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 206 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_14";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_15 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 207 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_15";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_16 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 208 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_16";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_17 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 209 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_17";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_18 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 210 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_18";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		icumx_cmd0_19 {
+			compatible = "generic-uio";
+			reg = <0 0xe6610000 0 0x8000>;
+			interrupts = <GIC_SPI 211 IRQ_TYPE_LEVEL_HIGH>;
+			linux,uio-name = "irq_icumx_cmd_00_19";
+			ipmmu-id = <RCAR_IPMMU_RT0 1>;
+			iommus = <&ipmmu_rt0 1>, <&ipmmu_rt0 2>;
+		};
+
+		chipid0: prr@fff00044 {
+			compatible = "generic-uio";
+			reg = <0 0xfff00044 0 0x1000>;
+			linux,uio-name = "chipid_00";
+		};
+
+		qos0: qos0@e67e0000 {
+			compatible = "generic-uio";
+			reg = <0 0xe67e0000 0 0x11000>;
+			linux,uio-name = "qos_00";
+		};
+
+		rcarbusmoni0: rcarbusmoni0@e67e0000 {
+			compatible = "generic-uio";
+			reg = <0 0xe67e0000 0 0x11000>;
+			linux,uio-name = "rcarbusmoni_00";
+		};
+
+		rcarbusmoni1: rcarbusmoni1@e6790000 {
+			compatible = "generic-uio";
+			reg = <0 0xe6790000 0 0x3800>;
+			linux,uio-name = "rcarbusmoni_01";
+		};
+
+		fbc0: fbc@ff830000 {
+			compatible = "renesas,uio-fbc";
+			reg = <0 0xff830000 0 0x2000>;
+			clocks = <&cpg CPG_CORE R8A779F0_CLK_POST>,
+				 <&cpg CPG_CORE R8A779F0_CLK_POST2>;
+			clock-names = "fbc_post", "fbc_post2";
+			interrupts = <GIC_SPI 740 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "fbc_00";
+		};
+
+		fba_ca55d0: fba@ff870000 {
+			compatible = "generic-uio";
+			reg = <0 0xff870000 0 0x1000>;
+			interrupts = <GIC_SPI 621 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "fba_ca55d0";
+		};
+
+		post_fba_ca55d0 {
+			compatible = "generic-uio";
+			reg = <0 0xff870000 0 0x1000>;
+			clocks = <&cpg CPG_CORE R8A779F0_CLK_POST2>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "post_fba_ca55d1";
+		};
+
+		fba_ca55d1: fba@ff871000 {
+			compatible = "generic-uio";
+			reg = <0 0xff871000 0 0x1000>;
+			interrupts = <GIC_SPI 624 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "fba_ca55d1";
+		};
+
+		post_fba_ca55d1 {
+			compatible = "generic-uio";
+			reg = <0 0xff871000 0 0x1000>;
+			clocks = <&cpg CPG_CORE R8A779F0_CLK_POST2>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "post_fba_ca55d1";
+		};
+
+		fba_ca550: fba@ff872000 {
+			compatible = "generic-uio";
+			reg = <0 0xff872000 0 0x1000>;
+			interrupts = <GIC_SPI 622 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "fba_ca550";
+		};
+
+		post_fba_ca550 {
+			compatible = "generic-uio";
+			reg = <0 0xff872000 0 0x1000>;
+			clocks = <&cpg CPG_CORE R8A779F0_CLK_POST2>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "post_fba_ca550";
+		};
+
+		fba_ca551: fba@ff873000 {
+			compatible = "generic-uio";
+			reg = <0 0xff873000 0 0x1000>;
+			interrupts = <GIC_SPI 623 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "fba_ca551";
+		};
+
+		post_fba_ca551 {
+			compatible = "generic-uio";
+			reg = <0 0xff873000 0 0x1000>;
+			clocks = <&cpg CPG_CORE R8A779F0_CLK_POST2>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "post_fba_ca551";
+		};
+
+		fba_ca552: fba@ff874000 {
+			compatible = "generic-uio";
+			reg = <0 0xff874000 0 0x1000>;
+			interrupts = <GIC_SPI 625 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "fba_ca552";
+		};
+
+		post_fba_ca552 {
+			compatible = "generic-uio";
+			reg = <0 0xff874000 0 0x1000>;
+			clocks = <&cpg CPG_CORE R8A779F0_CLK_POST2>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "post_fba_ca552";
+		};
+
+		fba_ca553: fba@ff875000 {
+			compatible = "generic-uio";
+			reg = <0 0xff875000 0 0x1000>;
+			interrupts = <GIC_SPI 626 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "fba_ca553";
+		};
+
+		post_fba_ca553 {
+			compatible = "generic-uio";
+			reg = <0 0xff875000 0 0x1000>;
+			clocks = <&cpg CPG_CORE R8A779F0_CLK_POST2>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "post_fba_ca553";
+		};
+
+		rfso0: rfso@ffe80000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe80000 0 0x1000>;
+			interrupts = <GIC_SPI 218 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_00";
+		};
+
+		rfso1: rfso@ffe81000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe81000 0 0x1000>;
+			interrupts = <GIC_SPI 219 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_01";
+		};
+
+		rfso2: rfso@ffe82000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe82000 0 0x1000>;
+			interrupts = <GIC_SPI 220 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_02";
+		};
+
+		rfso3: rfso@ffe83000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe83000 0 0x1000>;
+			interrupts = <GIC_SPI 221 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_03";
+		};
+
+		rfso4: rfso@ffe84000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe84000 0 0x1000>;
+			interrupts = <GIC_SPI 222 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_04";
+		};
+
+		rfso5: rfso@ffe85000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe85000 0 0x1000>;
+			interrupts = <GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_05";
+		};
+
+		rfso6: rfso@ffe86000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe86000 0 0x1000>;
+			interrupts = <GIC_SPI 224 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_06";
+		};
+
+		rfso7: rfso@ffe87000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe87000 0 0x1000>;
+			interrupts = <GIC_SPI 225 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_07";
+		};
+
+		rfso8: rfso@ffe88000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe88000 0 0x1000>;
+			interrupts = <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_08";
+		};
+
+		rfso9: rfso@ffe89000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe89000 0 0x1000>;
+			interrupts = <GIC_SPI 227 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_09";
+		};
+
+		rfso10: rfso@ffe8a000 {
+			compatible = "generic-uio";
+			reg = <0 0xffe8a000 0 0x1000>;
+			interrupts = <GIC_SPI 228 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			linux,uio-name = "rfso_10";
+		};
+
+		ufs: scsi@e6860000 {
+			compatible = "renesas,r8a779f0-ufs";
+			reg = <0 0xe6860000 0 0x100>;
+			interrupts = <GIC_SPI 235 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 1514>, <&ufs30_clk>;
+			clock-names = "fck", "ref_clk";
+			freq-table-hz = <200000000 200000000>, <38400000 38400000>;
+			power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
+			resets = <&cpg 1514>;
+			status = "disabled";
+		};
+	};
+
+	thermal-zones {
+		sensor_thermal1: sensor-thermal1 {
+			polling-delay-passive = <250>;
+			polling-delay = <0>;
+			thermal-sensors = <&tsc 0>;
+			//sustainable-power = <1000>;
+
+			trips {
+				sensor1_crit: sensor1-crit {
+					temperature = <120000>;
+					hysteresis = <1000>;
+					type = "critical";
+				};
+			};
+		};
+
+		sensor_thermal2: sensor-thermal2 {
+			polling-delay-passive = <250>;
+			polling-delay = <0>;
+			thermal-sensors = <&tsc 1>;
+			//sustainable-power = <1000>;
+
+			trips {
+				sensor2_crit: sensor2-crit {
+					temperature = <120000>;
+					hysteresis = <1000>;
+					type = "critical";
+				};
+			};
+		};
+
+		sensor_thermal3: sensor-thermal3 {
+			polling-delay-passive = <250>;
+			polling-delay = <0>;
+			thermal-sensors = <&tsc 2>;
+			//sustainable-power = <1000>;
+
+			trips {
+				sensor3_crit: sensor3-crit {
+					temperature = <120000>;
+					hysteresis = <1000>;
+					type = "critical";
+				};
+			};
 		};
 	};
 
 	timer {
 		compatible = "arm,armv8-timer";
-		interrupts-extended = <&gic GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>,
-				      <&gic GIC_PPI 14 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>,
-				      <&gic GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>,
-				      <&gic GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(1) | IRQ_TYPE_LEVEL_LOW)>;
+		interrupts-extended = <&gic GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(8) | IRQ_TYPE_LEVEL_LOW)>,
+				      <&gic GIC_PPI 14 (GIC_CPU_MASK_SIMPLE(8) | IRQ_TYPE_LEVEL_LOW)>,
+				      <&gic GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(8) | IRQ_TYPE_LEVEL_LOW)>,
+				      <&gic GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(8) | IRQ_TYPE_LEVEL_LOW)>;
 	};
 };
diff --git a/include/dt-bindings/ipmmu/rcar-ipmmu-domains.h b/include/dt-bindings/ipmmu/rcar-ipmmu-domains.h
new file mode 100644
index 0000000000..266c4b5b1b
--- /dev/null
+++ b/include/dt-bindings/ipmmu/rcar-ipmmu-domains.h
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * Copyright (C) 2020 Cogent Embedded Inc.
+ * Copyright (C) 2020 Renesas Electronics Corporation
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+#ifndef __LINUX_RCAR_IPMMU_DOMAINS_H__
+#define __LINUX_RCAR_IPMMU_DOMAINS_H__
+
+#define RCAR_IPMMU_INVALID	0
+#define RCAR_IPMMU_MM		1
+#define RCAR_IPMMU_VI0		2
+#define RCAR_IPMMU_VI1		3
+#define RCAR_IPMMU_VP0		4
+#define RCAR_IPMMU_VP1		5
+#define RCAR_IPMMU_VC0		6
+#define RCAR_IPMMU_VC1		7
+#define RCAR_IPMMU_PV0		8
+#define RCAR_IPMMU_PV1		9
+#define RCAR_IPMMU_PV2		10
+#define RCAR_IPMMU_PV3		11
+#define RCAR_IPMMU_IR		12
+#define RCAR_IPMMU_HC		13
+#define RCAR_IPMMU_RT0		14
+#define RCAR_IPMMU_RT1		15
+#define RCAR_IPMMU_MP		16
+#define RCAR_IPMMU_DS0		17
+#define RCAR_IPMMU_DS1		18
+#define RCAR_IPMMU_VIP0		19
+#define RCAR_IPMMU_VIP1		20
+#define RCAR_IPMMU_3DG		21
+#define RCAR_IPMMU_VC		22
+
+#endif /* __LINUX_RCAR_IPMMU_DOMAINS_H__ */
+
diff --git a/include/dt-bindings/ipmmu/rcar-ipmmu.h b/include/dt-bindings/ipmmu/rcar-ipmmu.h
new file mode 100644
index 0000000000..231d350c76
--- /dev/null
+++ b/include/dt-bindings/ipmmu/rcar-ipmmu.h
@@ -0,0 +1,15 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * Copyright (C) 2020 Cogent Embedded Inc.
+ * Copyright (C) 2020 Renesas Electronics Corporation
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+#ifndef _DT_BINDINGS_RCAR_IPMMU_H
+#define _DT_BINDINGS_RCAR_IPMMU_H
+
+#include "rcar-ipmmu-domains.h"
+
+#endif /* _DT_BINDINGS_RCAR_IPMMU_H */
-- 
2.30.2

